### ISSUE TITLE
Should document the case train_set is specified in valid_sets of lightgbm.train

### ISSUE THREAD

**Issue by fujii** [Microsoft/LightGBM#1374]
In the document,

http://lightgbm.readthedocs.io/en/latest/Python-API.html#training-api

> If there’s more than one, will check all of them. 

This is not correct. Looking into the source code, train_set is handled differently.

https://github.com/Microsoft/LightGBM/blob/master/python-package/lightgbm/engine.py#L136

**Comment by guolinke** [comment]
@fujii 
Sorry, I didn't got your point. what is the problem ? 

This handling is for efficiency: it reuses the train-set if it is listed in valid_sets.

**Comment by fujii** [comment]
Thank you for your feedback, @guolinke .
The document of `early_stopping_rounds` says it will check *all* of `valid_names`. But, it doesn't check the dataset specified in `train_set`.

**Comment by guolinke** [comment]
okay, got it.
Actually, using early stopping over training set is not reasonable, since the loss over training set will always be reduced.
But you are right, we should point it out in document. And you are welcomed to create a PR for the changes.

**Comment by fujii** [comment]
Thanks. I will try.

**Comment by StrikerRUS** [comment]
Fixed in #1393.
